ipa-core/src/protocol/basics/reshare.rs

```diff
@@ -212,21 +212,25 @@ mod tests {
                     Reshare,
                 },
                 context::{
-                    Context, SemiHonestContext, UpgradableContext, UpgradedContext,
-                    UpgradedMaliciousContext, Validator,
+                    Context, SemiHonestContext, SpecialAccessToUpgradedContext, UpgradableContext,
+                    UpgradedContext, UpgradedMaliciousContext, Validator,
                 },
                 prss::SharedRandomness,
                 RecordId,
             },
             rand::{thread_rng, Rng},
             secret_sharing::{
                 replicated::{
-                    malicious::{AdditiveShare as MaliciousReplicated, ExtendableField},
+                    malicious::{
+                        AdditiveShare as MaliciousReplicated, ExtendableField,
+                        ThisCodeIsAuthorizedToDowngradeFromMalicious,
```

Comment:
why did you need to add `ThisCodeIsAuthorizedToDowngradeFromMalicious`?

Reply:
That's just to allow calling `downgrade` method manually